### PR TITLE
OCEANWATER-621 Implement SOC Functionality & GSAP Tagged Branch

### PR DESCRIPTION
**Pull request by jfugate1**
## Linked Issues:
| Jira Ticket 🎟️   | [Oceanwater-621](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-621) |
| ----------- | ----------- |
| EPIC ⚡| [Oceanwater-84](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-84) |
| Github :octocat:  | # |

## Summary of Changes
* Implemented SOC functionality
* Updated Setup Dev Environment Instructions

## Test
* Replace GSAP repository with v1.0-OW
* Recompile GSAP
* Build/Run OW
* Subscribe to power_system_node/state_of_charge rostopic

### CODE REVIEW

File: ow_power_system/src/power_system_node.cpp [hunk 1]
```diff
@@ -109,22 +109,22 @@ int main(int argc, char* argv[]) {
...
+      UData currentSOC = eod_event.getState()[0];
       // For this example, we will print the median SOC
-      //auto samples2 = currentSOC.getVec();
-      //std::sort(samples2.begin(), samples2.end());
-      //double soc_median = samples2.at(samples2.size() / 2);
-      //soc_msg.data = soc_median;
+      auto samplesSOC = currentSOC.getVec();
+      std::sort(samplesSOC.begin(), samplesSOC.end());
+      double soc_median = samplesSOC.at(samplesSOC.size() / 2);
+      soc_msg.data = soc_median;
```
Comment (Samahu) [inline ow_power_system/src/power_system_node.cpp:127]: Shouldn't these lines be within the else part of `if (eod_time.uncertainty() != UType::Samples) {`

Comment (Samahu) [reply ow_power_system/src/power_system_node.cpp:127]: I am just asking if the sate_of_charge can be affected by the uncertainty

Comment (jfugate1) [reply ow_power_system/src/power_system_node.cpp]: Yes. I should have moved this code... Should be resolved now.

**Review by Samahu — APPROVED** [review]
Works as described!

**Comment by kmdalal** [comment]
I'm taking myself off this review because I'm too buried and I think 1-2 reviewers is sufficient for this one (and most PR's actually).   I did a quick scan of the changed files and it looked fine.

**Pull request merged**